tests/selenium/servers.py

```diff
@@ -15,62 +18,66 @@ def toggle_server_section(self, name):
+        reply_to.send_keys('test@localhost')
+        signature = self.by_name('srv_setup_stepper_profile_signature')
+        signature.send_keys('Test')
+        self.by_id('step_config_action_finish').click()
+        wait = WebDriverWait(self.driver, 30)
+        element = wait.until(EC.visibility_of_element_located((By.CLASS_NAME, "sys_messages")))
+        sys_message_text = element.text
+        sys_message_texts = sys_message_text.split('\n')
+        assert any("Authentication failed" in text for text in sys_message_texts), "Authentication failed message not found"
 
-    def smtp_del(self):
-        self.by_class('delete_smtp_connection').click()
-        self.confirm_alert()
-        self.wait_on_sys_message()
-        assert self.by_class('sys_messages').text == 'Server deleted'
-        self.toggle_server_section('smtp')
-
-    def imap_add(self):
-        self.toggle_server_section('imap')
-        name = self.by_name('new_imap_name')
-        name.send_keys('Test')
-        addr = self.by_name('new_imap_address')
-        addr.send_keys('localhost')
-        port = self.by_name('new_imap_port')
-        port.clear()
-        port.send_keys(143)
-        self.by_id('imap_notls').click()
-        self.by_name('submit_imap_server').click()
-        self.wait_on_sys_message()
-        assert self.by_class('sys_messages').text == 'Added server!' or self.by_class('sys_messages').text == 'Could not add server: Connection refused'
-
-    def imap_confirm(self):
-        user = self.by_id('imap_user_'+IMAP_ID)
-        user.send_keys('testuser')
-        passw = self.by_id('imap_pass_'+IMAP_ID)
-        passw.send_keys('testuser')
-        self.by_class('test_imap_connect').click()
-        self.wait_on_sys_message()
-        assert self.by_class('sys_messages').text == 'Successfully authenticated to the IMAP server'
-        self.toggle_server_section('imap')
+    def server_jmap_add(self):
+        jmap_checkbox = self.by_id('srv_setup_stepper_only_jmap')
+        jmap_checkbox.click()
+        jmap_address = self.by_name('srv_setup_stepper_jmap_address')
+        jmap_address.send_keys('jmap.server.com')
+        finish_button = self.by_id('step_config_action_finish')
+        self.driver.execute_script("arguments[0].click(); return false;", finish_button)
+        wait = WebDriverWait(self.driver, 30)
+        element = wait.until(EC.visibility_of_element_located((By.CLASS_NAME, "sys_messages")))
+        sys_message_text = element.text
+        sys_message_texts = sys_message_text.split('\n')
+        assert any("Authentication failed" in text for text in sys_message_texts), "Authentication failed message not found"
 
 if __name__ == '__main__':
 
     print("SERVERS TEST")
     test_runner(ServersTest, [
         'load_servers_page',
-        'smtp_add',
-        'smtp_del',
-        'imap_add',
-        #'imap_confirm'
+        'server_stmp_and_imap_add',
+        'server_jmap_add'
```

Comment:
With the updated testing structure for delete, forget, and test functionalities, it's imperative to have a connected server During testing, these buttons are only visible when a server is connected. Without a server connection, we cannot proceed to the next steps as the buttons associated with these functionalities may not be visible. Ensuring server connectivity is essential for successful testing and we are using localhost to connect `smtp,imap and jmap`. it will not connect we will get error message: "Authentication failed"

Reply:
Yes, I see. There are two ways to make this happen:
- let user configure selenium tests with their own (real) server account to run the tests with
- make sure the workflows added by @wangxiaoerYah in https://github.com/cypht-org/cypht/pull/954 run these tests with an actual server they setup locally in the docker container

In both cases, I think a user-configurable server just for the tests is a good idea - maybe through .env or other config files in the tests namespace.